lib/Doctrine/Annotations/Metadata/Builder/PropertyMetadataBuilder.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Annotations\Metadata\Builder;
+
+use Doctrine\Annotations\Metadata\PropertyMetadata;
+
+/**
+ * @internal
+ */
+final class PropertyMetadataBuilder
+{
+    /** @var string */
+    private $name;
+
+    /** @var string[]|null */
+    private $type;
+
+    /** @var bool */
+    private $required = false;
+
+    /** @var bool */
+    private $default = false;
+
+    /** @var mixed[]|null */
```

Comment:
Is it possible to have a stricter type declaration here? Specifically, do we allow objects and resources? I think we only allow constant-compatible values here.